File: process-api/src/main/java/br/com/softplan/process/api/service/ProcessService.java
package br.com.softplan.process.api.service;

import org.springframework.data.domain.Page;

import br.com.softplan.process.api.entity.Process;
import br.com.softplan.process.api.entity.ProcessReview;

public interface ProcessService {

	Process createOrUpdate(Process process);

	Process findById(String id);

	void delete(String id);

	Page<Process> listProcess(int Page, int count);

	ProcessReview createProcessReview(ProcessReview processReview);
		
	Iterable<ProcessReview> listProcessReview(String processId);
	
	Page<Process> findByCurrentUser(int page, int count, String userId);

	Page<Process> findByParameters(int page, int count, String subject, String status, String priority);

	Page<Process> findByParametersAndCurrentUser(int page, int count, String subject, String status, String priority,
			String userId);

	Page<Process> findByNumber(int page, int count, Integer number);

	Iterable<Process> findAll();
		
	Page<Process> findByParameterAndAssignedUser(int page, int count, String subject, String status, String priority,
			String assignedUser);
}

File: frontend/processos-front-/src/app/services/process/process.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from '@angular/common/http';
import { PROCESSOS_API } from '../processos.api';
import { Process } from '../../model/Process';

@Injectable()
export class ProcessService {

  constructor(private http: HttpClient) {}

  createOrUpdate(process: Process){
    if(process.id != null && process.id != ''){
      return this.http.put(`${PROCESSOS_API}/api/process`,process);
    } else {
      process.id = null;
      return this.http.post(`${PROCESSOS_API}/api/process`, process);
    }
  }

  findAll(page:number,count:number){
    return this.http.get(`${PROCESSOS_API}/api/process/${page}/${count}`);
  }

  findById(id:string){
    return this.http.get(`${PROCESSOS_API}/api/process/${id}`);
  }

  delete(id:string){
    return this.http.delete(`${PROCESSOS_API}/api/process/${id}`);
  }

  findByParams(page:number,count:number,t:Process){
    t.number = t.number == null ? 'uninformed' : t.number;
    return this.http.get(`${PROCESSOS_API}/api/process/${page}/${count}/${t.number}`);
  }


}